### PR TITLE
date-picker extend options

### PR DESCRIPTION
**Pull request by qubaomingg**
In my business scene I need to format the `time-picker` of `date-picker` when option `showTime` is setted.

And I found the scene is a general demand, eg [#771](https://github.com/ant-design/ant-design/issues/771).

So, I send this PR, please think about it.

### CODE REVIEW

**Comment by afc163** [comment]
Extend the `showTime` prop is better.

```jsx
<DatePicker showTime={{ format: ..., disabledHours: ... }}>
```

**Comment by freestyle21** [comment]
@afc163 your suggest is better. and I have modified, please review ~

File: components/date-picker/wrapPicker.jsx [hunk 1]
```diff
@@ -94,12 +94,22 @@ export default function wrapPicker(Picker, defaultFormat) {
...
+        showHour: timeFormat && timeFormat.indexOf('HH') >= 0
+      };
       const timePicker = props.showTime ? (
         <TimePicker
           prefixCls="ant-time-picker"
           placeholder={locale.timePickerLocale.placeholder}
           locale={locale.timePickerLocale}
-          transitionName="slide-up" />
+          transitionName="slide-up"
+          {...props.showTime}
```
Comment (benjycui) [inline components/date-picker/wrapPicker.jsx]: Will be better.

```jsx
        <TimePicker
          {...props.showTime}
          {...rcTimePickerProps}
          prefixCls="ant-time-picker"
          placeholder={locale.timePickerLocale.placeholder}
          locale={locale.timePickerLocale}
          transitionName="slide-up"
        />
```

**Comment by afc163** [comment]
可以绑定下邮箱信息。

File: components/date-picker/index.md [hunk 2]
```diff
@@ -34,9 +34,9 @@ english: DatePicker
 | popupStyle   | 格外的弹出日历样式   | object     | {}   |
 | size         | 输入框大小，`large` 高度为 32px，`small` 为 22px，默认是 28px | string   | 无  |
 | locale       | 国际化配置 | object   | [默认配置](https://github.com/ant-design/ant-design/issues/424)  |
-| showTime     | 增加时间选择功能 | bool | false  |
 | onOk         | 点击确定按钮的回调 | function(Date value) | 无 |
 | getCalendarContainer | 定义浮层的容器，默认为 body 上新建 div | function(trigger) | 无 |
+| showTime     | 增加时间选择功能  | object | [TimePicker Options](http://ant.design/components/time-picker/#api) |
```
Comment (afc163) [inline components/date-picker/index.md]: object or bool

File: components/date-picker/wrapPicker.jsx [hunk 3]
```diff
@@ -94,12 +94,22 @@ export default function wrapPicker(Picker, defaultFormat) {
...
+      const timeFormat = props.showTime && props.showTime.format;
+      const rcTimePickerProps = {
+        formatter: new DateTimeFormat(timeFormat || 'HH:mm:ss'),
+        showSecond: timeFormat && timeFormat.indexOf('ss') >= 0,
+        showHour: timeFormat && timeFormat.indexOf('HH') >= 0
+      };
       const timePicker = props.showTime ? (
         <TimePicker
+          {...props.showTime}
+          {...rcTimePickerProps}
```
Comment (afc163) [inline components/date-picker/wrapPicker.jsx]: props.showTime 放在 rcTimePickerProps 下面如何？

这样用户可以强制覆盖默认的 showSecond showHour 属性。

**Comment by afc163** [comment]
还有 RangePicker 的 showTime 也同样支持下？麻烦了~

**Pull request merged**